gen_metal3_config.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/bash
+
+set -e
+
+script_name=$0
+
+RHCOS_IMAGE_URL=""
+PROVISIONING_INTERFACE="eno1"
+PROVISIONING_ADDRESS="172.22.0.3"
+
+function usage {
+    cat - <<EOF
+
+$script_name [-h|-u URL [-i INTERFACE] [-a IPADDR]]
+
+    -h            Output this help text.
+
+    -a IPADDR     Specify the VIP address where the provisioning service
+                  should listen.
+    -i INTERFACE  Specify the network interface on the provisioning net.
+                  Defaults to "eno1".
+    -u URL        Specify the RHCOS image URL to use to prime the cache.
+
+EOF
+}
+
+while getopts "hi:u:" opt; do
+    case ${opt} in
+        h)
+            usage;
+            exit 1
+            ;;
+        i)
+            PROVISIONING_INTERFACE=$OPTARG
+            ;;
+        u)
+            RHCOS_IMAGE_URL=$OPTARG
+            ;;
+        a)
+            PROVISIONING_ADDRESS=$OPTARG
```

Comment:
Interesting.. I'm not sure that provisioning address is really very useful as a configurable.  I was thinking we should specify the cidr or even just the network and extrapolate the rest from that.  eg with a setting 172.22.0.0 we could set the cidr, the provisioning IP and the dhcp range.

Reply:
Is it safe to assume a particular IP, even if we know the CIDR?

Reply:
Yes.  We control the network.. so long as the network is ok for the operator to use we are golden.  We can divide the IP space up as we see fit.

Reply:
@imain want to work on that as a follow-up?  This script seems like a good start to me

Reply:
yeah I'll look at doing that.